python/hail/docs/hailpedia/matrixtable.rst

```diff
@@ -0,0 +1,404 @@
+
+    >>> filt_mt = mt.filter_rows(hl.len(mt.alleles) == 2)
+
+    >>> filt_mt = mt.filter_cols(hl.agg.mean(mt.GQ) < 20)
+
+    >>> filt_mt = mt.filter_entries(mt.DP < 5)
+
+These expressions can compute arbitrarily over the data: the :meth:`.MatrixTable.filter_cols`
+example above aggregates entries per column of the matrix table to compute the
+mean of the `GQ` field, and removes columns where the result is smaller than 20.
+
+**Annotate**
+
+:class:`.MatrixTable` has four methods to add new fields or update existing fields:
+
+- :meth:`.MatrixTable.annotate_rows`
+- :meth:`.MatrixTable.annotate_cols`
+- :meth:`.MatrixTable.annotate_entries`
+- :meth:`.MatrixTable.annotate_globals`
+
+Annotate methods take keyword arguments where the key is the name of the new
+field to add and the value is an expression specifying what should be added.
+
+The simplest example is adding a new global field `foo` that just contains the constant
+5.
+
+    >>> mt_new = mt.annotate_globals(foo = 5)
+    >>> print(mt.globals.dtype.pretty())
+    struct {
+        foo: int32
+    }
+
+Another example is adding a new row field `call_rate` which computes the fraction
+of non-missing entries `GT` per row:
+
+    >>> mt_new = mt.annotate_rows(call_rate = hl.agg.fraction(hl.is_defined(mt.GT)))
+
+Annotate methods are also useful for updating values. For example, to update the
+GT entry field to be missing if `GQ` is less than 20, we can do the following:
+
+    >>> mt_new = mt.annotate_entries(GT = hl.case()
+    ...                                     .when(mt.GQ >= 20, mt.GT)
+    ...                                     .or_missing())
+
+**Select**
+
+Select is used to create a new schema for a dimension of the matrix table. For
+example, following the matrix table schemas from importing a VCF file (shown above),
+to create a hard calls dataset where each entry only contains the `GT` field
+one can do the following:
+
+    >>> mt_new = mt.select_entries('GT')
+    >>> print(mt_new.entry.dtype.pretty())
+    struct {
+        GT: call
+    }
+
+:class:`.MatrixTable` has four select methods that select and create new fields:
+
+- :meth:`.MatrixTable.select_rows`
+- :meth:`.MatrixTable.select_cols`
+- :meth:`.MatrixTable.select_entries`
+- :meth:`.MatrixTable.select_globals`
+
+Each method can take either strings referring to top-level fields, an attribute
+reference (useful for accessing nested fields), as well as keyword arguments
+``KEY=VALUE`` to compute new fields. The Python unpack operator ``**`` can be
+used to specify that all fields of a Struct should become top level fields.
+However, be aware that all top-level field names must be unique. In this
+example, `**mt['info']` would fail because `DP` already exists as an entry
+field.
+
+The example below will keep the row keys `locus` and `alleles` as well as add
+two new fields: `AC` is making the subfield `AC` into a top level field and
+`n_filters` is a new computed field.
```

Comment:
this sentence reads weird. maybe: 

> as well as add two new fields. ``AC = mt.info.AC`` is making the subfield ``AC`` into a top level field. The second keyword argument is computing the new field `n_filters`.

eh. not sure any further explanation after "add two new fields" is necessary.